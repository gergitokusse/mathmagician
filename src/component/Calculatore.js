import MyButtons from './CalculatoreButtons';

function Mycalculator() {
  return (
    <div className="container">
      <div>
        <h1>
          Letus do same think new
        </h1>
      </div>
      <MyButtons />
    </div>
  );
}

export default Mycalculator;
