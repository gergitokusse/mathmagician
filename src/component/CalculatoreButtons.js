import React, { useState, useEffect } from 'react';
import calculate from '../logic/calculate';

const MyButtons = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  useEffect(() => {
    const MyButtons = document.querySelectorAll('.btn');
    const btnEventHandler = (e) => {
      const response = calculate({ total, next, operation }, e.target.innerText);
      setTotal(response.total);
      setNext(response.next);
      setOperation(response.operation);
    };

    MyButtons.forEach((button) => {
      button.addEventListener('click', btnEventHandler);
    });

    return () => {
      MyButtons.forEach((button) => {
        button.removeEventListener('click', btnEventHandler);
      });
    };
  }, [total, next, operation]);
  return (
    <div className="calbutton">
      <div className="type-area">
        {total || 0}
        { operation }
        { next }
      </div>
      <button className="btn" type="button">AC</button>
      <button className="btn" type="button">+/-</button>
      <button className="btn" type="button">%</button>
      <button className="btn opration" type="button">รท</button>
      <button className="btn" type="button">7</button>
      <button className="btn" type="button">8</button>
      <button className="btn" type="button">9</button>
      <button className="btn opration" type="button">x</button>
      <button className="btn" type="button">4</button>
      <button className="btn" type="button">5</button>
      <button className="btn" type="button">6</button>
      <button className="btn opration" type="button">-</button>
      <button className="btn" type="button">1</button>
      <button className="btn" type="button">2</button>
      <button className="btn" type="button">3</button>
      <button className="btn opration" type="button">+</button>
      <button className="btn zero" type="button">0</button>
      <button className="btn" type="button">.</button>
      <button className="btn opration" type="button">=</button>
    </div>
  );
};

export default MyButtons;
