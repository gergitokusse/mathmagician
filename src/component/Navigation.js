// eslint-disable-next-line
import { Link } from 'react-router-dom';

function Navigation() {
  return (
    <div className="navigation">
      <h3>Math Magicians</h3>
      <ul>
        <li>
          <Link to="/Home">
            Home
          </Link>
        </li>
        <li>
          <Link to="/Calculatore">Calculatore</Link>
        </li>
        <li className="last">
          <Link to="/Quete">Quete</Link>
        </li>
      </ul>
    </div>
  );
}

export default Navigation;
