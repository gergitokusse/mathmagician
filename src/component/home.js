function Home() {
  return (
    <div>
      <h1>Welcome to Our Page</h1>
      <p>
        What is Lorem Ipsum? Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industrys standard dummy text ever since the
        1500s, when an unknown printer took a galley of type and scrambled it to make a
        type specimen book.
      </p>
      <p>
        What is Lorem Ipsum? Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industrys standard dummy text ever since the
        1500s, when an unknown printer took a galley of type and scrambled it to make a
        type specimen book.
      </p>
    </div>
  );
}

export default Home;
