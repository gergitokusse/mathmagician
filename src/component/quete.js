import { Qutes } from './CalculatoreButtons';

function Quete() {
  return (
    <div>
      <Qutes />
    </div>
  );
}

export default Quete;
