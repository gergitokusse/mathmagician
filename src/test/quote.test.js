import { render, screen } from '@testing-library/react';
import { Qutes } from '../component/CalculatoreButtons';

describe('Test Quote Component', () => {
  test('Text content testing', () => {
    render(<Qutes />);
    const pElement = screen.getByText(/Pls wait moment it is Loading.../i);
    expect(pElement).toMatchSnapshot();
  });
});
