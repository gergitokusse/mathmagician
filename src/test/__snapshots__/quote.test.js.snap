// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quote Component Text content testing 1`] = `
<div
  class="loading"
>
   Pls wait moment it is Loading... 
</div>
`;
